modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -274,10 +280,16 @@ def srvsvc_netshareenum(ip)
       name    = Rex::Text.to_ascii(name)
       s_type  = Rex::Text.to_ascii(share_type(types[t]))
       comment = Rex::Text.to_ascii(comment)
-
+      if datastore['Check_Write_Access']
```

Comment:
You're checking write access only when using SRVSVC, which is unexpected I think.